src/HtmlGenerator/CommandLineOptions.cs

```diff
@@ -0,0 +1,293 @@
+                    }
+                }
+
+                if (arg == "/assemblylist")
+                {
+                    emitAssemblyList = true;
+                    continue;
+                }
+
+                if (string.Equals(arg, "/donotincludereferencedprojects", StringComparison.OrdinalIgnoreCase))
+                {
+                    doNotIncludeReferencedProjects = true;
+                    continue;
+                }
+
+                if (arg == "/nobuiltinfederations")
+                {
+                    noBuiltInFederations = true;
+                    Log.Message("Disabling built-in federations.");
+                    continue;
+                }
+
+                if (arg.StartsWith("/federation:", StringComparison.Ordinal))
+                {
+                    string server = arg.Substring("/federation:".Length);
+                    Log.Message($"Adding federation '{server}'.");
+                    federations.Add(server);
+                    continue;
+                }
+
+                if (arg.StartsWith("/offlinefederation:", StringComparison.Ordinal))
+                {
+                    var match = Regex.Match(arg, "/offlinefederation:(?<server>[^=]+)=(?<file>.+)");
+                    if (match.Success)
+                    {
+                        var server = match.Groups["server"].Value;
+                        var assemblyListFileName = match.Groups["file"].Value;
+                        offlineFederations[server] = assemblyListFileName;
+                        Log.Message($"Adding federation '{server}' (offline from '{assemblyListFileName}').");
+                    }
+                    continue;
+                }
+
+                if (string.Equals(arg, "/noplugins", StringComparison.OrdinalIgnoreCase))
+                {
+                    loadPlugins = false;
+                    continue;
+                }
+
+                if (string.Equals(arg, "/useplugins", StringComparison.OrdinalIgnoreCase))
+                {
+                    loadPlugins = true;
+                    continue;
+                }
+
+                if (arg.StartsWith("/noplugin:", StringComparison.Ordinal))
+                {
+                    pluginBlacklist.Add(arg.Substring("/noplugin:".Length));
+                    continue;
+                }
+
+                if (arg == "/excludetests")
+                {
+                    excludeTests = true;
+                    continue;
+                }
+
+                if (arg.StartsWith("/root:", StringComparison.Ordinal))
+                {
+                    rootPath = Path.GetFullPath(arg.Substring("/root:".Length).StripQuotes());
+                    continue;
+                }
+
+                try
+                {
+                    AddProject(projects, arg);
+                }
+                catch (Exception ex)
+                {
+                    Log.Write("Exception: " + ex, ConsoleColor.Red);

```

Comment:
I didn't extract the `Log.Write`/`Message`/`Exception calls. Could be done when tests start caring.